models/vrp.rb

```diff
@@ -462,6 +466,30 @@ def self.deduce_unavailable_days(hash, element, start_index, end_index, type)
       collect_unavaible_day_indices(element, start_index, end_index)
     end
 
+    def self.deduce_possible_days(hash, element, start_index)
+      convert_possible_dates_into_indices(element, hash, start_index)
+
+      element[:first_possible_days] = element[:first_possible_day_indices].to_a.slice(0..(element[:visits_number] || 1) - 1)
+      element[:last_possible_days] = element[:last_possible_day_indices].to_a.slice(0..element[:visits_number] || 1 - 1)
```

Comment:
```suggestion
      element[:last_possible_days] = element[:last_possible_day_indices].to_a.slice(0..(element[:visits_number] || 1) - 1)
```
If we use parenthesis above, we should use it here too